server/src/main/java/org/opensearch/common/unit/Fuzziness.java

```diff
@@ -189,7 +189,12 @@ public static Fuzziness parse(XContentParser parser) throws IOException {
                             return build(fuzziness);
                     }
                 } catch (NumberFormatException ex) {
-                    return build(fuzziness);
+                    try {
+                        final float minimumSimilarity = Float.parseFloat(fuzziness);
+                        return build(fuzziness);
+                    } catch (NumberFormatException e) {
+                        throw new IllegalArgumentException("Invalid fuzziness value: " + fuzziness);
```

Comment:
Let's leave a comment here to explain what it's validating.

Reply:
added in [8542176](https://github.com/opensearch-project/OpenSearch/pull/5805/commits/8542176477525291c2f4c1b6ab5c042537a7eaad)